http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -602,6 +608,29 @@ public void testResponseStatusCode()
         }
     }
 
+    @Test
+    public void testExecuteStreaming()
+            throws Exception
+    {
+        try (CloseableTestHttpServer server = newServer()) {
+            server.servlet().setResponseBody(LARGE_CONTENT);
+
+            Request request = prepareGet()
+                    .setUri(server.baseURI())
+                    .build();
+
+            executeRequest(server, request).ifPresent(streamingResponse -> {
+                try (streamingResponse) {
+                    String responseString = new String(ByteStreams.toByteArray(streamingResponse.getInputStream()), UTF_8);
+                    assertThat(responseString).isEqualTo(LARGE_CONTENT);
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
After the change to `StreamingResponse`:
```suggestion
                catch (IOException e) {
                    throw new UncheckedIOException(e);
```